src/librustc_middle/ty/layout.rs

```diff
@@ -1901,11 +1903,9 @@ impl<'tcx> LayoutOf for LayoutCx<'tcx, TyCtxt<'tcx>> {
 
     /// Computes the layout of a type. Note that this implicitly
     /// executes in "reveal all" mode.
-    fn layout_of(&self, ty: Ty<'tcx>) -> Self::TyAndLayout {
-        let param_env = self.param_env.with_reveal_all_normalized(self.tcx);
-        let ty = self.tcx.normalize_erasing_regions(param_env, ty);
-        let layout = self.tcx.layout_raw(param_env.and(ty))?;
-        let layout = TyAndLayout { ty, layout };
+    fn layout_of(&self, mut ty: Ty<'tcx>) -> Self::TyAndLayout {
+        ty = self.tcx.erase_regions(&ty);
+        let layout = self.tcx.layout_raw(self.param_env.and(ty))?;
```

Comment:
It's kind of silly but I've started doing this (cc @nikomatsakis):
```rust
tcx.erase_regions(&param_env.and(tcx.erase_regions(&foo)))
```

The inner `erase_regions` helps `ParamEnv` get rid of bounds when `foo` mentions lifetime parameters but not type/const ones. The outer `erase_regions` erases all of the bounds in the `ParamEnv` (if any remain).